src/streaming/controllers/BufferController.js

```diff
@@ -201,12 +204,12 @@ function BufferController(config) {
             logger.info('Append Init fragment', type, ' with representationId:', chunk.representationId, ' and quality:', chunk.quality, ', data size:', chunk.bytes.byteLength);
             appendToBuffer(chunk);
         } else {
-            eventBus.trigger(Events.INIT_REQUESTED, { sender: instance });
+            eventBus.trigger(Events.INIT_REQUESTED, { mediaType: type, sender: instance });
         }
     }
 
     function onMediaFragmentLoaded(e) {
-        if (e.fragmentModel !== streamProcessor.getFragmentModel()) return;
+        if (e.chunk.streamId !== streamId || e.chunk.mediaInfo.type != type) return;
```

Comment:
Suggest to do strong comparison: e.chunk.mediaInfo.type !== type

Reply:
of course, thanks, I missed it. I've updated